### PR TITLE
Fix external open state

### LINKED ISSUE #2535: Combobox with external open state does not handle keyboard events correctly
Steps to reproduce:
* configure Combobox with `isOpen` property and appropriate event handlers.
* click inside the combobox.
* move down with `DOWN` key.

Bug 1:
Cannot move up again with `UP` key.

Bug 2: 
If moving further down with more than 6 options, the view does not scroll.

### PR DESCRIPTION
**Pull request by kuhmuh**
Fixes #2535, #2536 

### Additional description

This PR belongs to handling with customized comboboxes. As I heavily use them in my project, I need this fixes, because the bugs are quite annoying.
---

### CONTRIBUTOR checklist (do not remove)
Please complete for every pull request

* [x] First-time contributors should sign the Contributor License Agreement. It's a fancy way of saying that you are giving away your contribution to this project. If you haven't before, wait a few minutes and a bot will comment on this pull request with instructions.
* [x] `npm run lint:fix` has been run and linting passes.
* [x] Mocha, Jest (Storyshots), and `components/component-docs.json` CI tests pass (`npm test`).
* [x] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [x] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [x] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [x] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).

### REVIEWER checklist (do not remove)

* [ ] TravisCI tests pass. This includes linting, Mocha, Jest, Storyshots, and `components/component-docs.json` tests.
* [ ] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [ ] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [ ] The Accessibility panel of each Storybook story has 0 violations (aXe). Open [http://localhost:9001/](http://localhost:9001/).
* [ ] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [ ] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).
###### Required only if there are markup / UX changes
* [ ] Add year-first date and commit SHA to `last-slds-markup-review` in `package.json` and push.
* [ ] Request a review of the deployed Heroku app by the Salesforce UX Accessibility Team.
* [ ] Add year-first review date, and commit SHA, `last-accessibility-review`, to `package.json` and push.
* [ ] While the contributor's branch is checked out, run `npm run local-update` within locally cloned [site repo](https://github.com/salesforce-ux/design-system-react-site) to confirm the site will function correctly at the next release.

### CODE REVIEW

File: components/combobox/combobox.jsx [hunk 1]
```diff
@@ -459,7 +459,11 @@ class Combobox extends React.Component {
 			}
 		} else if (this.props.isOpen !== nextProps.isOpen) {
 			// eslint-disable-next-line react/no-did-update-set-state
-			this.setState({ isOpen: nextProps.isOpen });
+				this.setState({
```
Comment (garygong) [inline components/combobox/combobox.jsx:462]: I think we only want to make this `setState` call only when closing the popover, so something like `if (!nextProps.isOpen) { ....`

File: components/combobox/combobox.jsx [hunk 2]
```diff
@@ -970,10 +974,8 @@ class Combobox extends React.Component {
 	handleRequestClose = (event, data) => {
 		if (this.props.events.onRequestClose) {
 			this.props.events.onRequestClose(event, data);
-		}
-
-		if (this.getIsOpen()) {
-			this.setState({ isOpen: false });
+		} else {
```
Comment (garygong) [inline components/combobox/combobox.jsx:977]: I don't think we should be limiting calling `handleClose` to only when `onRequestClose` is provided.

Instead, does this work?
```
if (this.props.events.onRequestClose) {
	this.props.events.onRequestClose(event, data);
}
if (this.getIsOpen()) {
	this.handleClose(event, { trigger: 'cancel' });
}
```

File: components/combobox/combobox.jsx (hunk 1 above)
Comment (kuhmuh) [reply components/combobox/combobox.jsx:462]: Ok. I can do this more explicit with a dedicated `if` statement.

Comment (kuhmuh) [reply components/combobox/combobox.jsx:462]: I retested behaviour

File: components/combobox/combobox.jsx (hunk 2 above)
Comment (kuhmuh) [reply components/combobox/combobox.jsx:977]: This should also work. I change this accordingly.

**Comment by kuhmuh** [comment]
I changed code accordingly and left some additional comments.
Thanks for reviewing.

**Comment by kuhmuh** [comment]
Can this proceed now after my changes?

**Comment by garygong** [comment]
Sorry for the delay.  Will take a look at this tomorrow.

**Comment by garygong** [comment]
LGTM! (I pushed a commit to remove the unrelated changes in `component-docs.json` and `package.json.)  Thanks a lot!

**Pull request merged**